crates/libcontainer/src/container/init_builder.rs

```diff
@@ -87,6 +87,11 @@ impl InitContainerBuilder {
         };
 
         let user_ns_config = UserNamespaceConfig::new(&spec)?;
+
+        if utils::rootless_required() && !utils::is_in_new_userns() && user_ns_config.is_none() {
```

Comment:
Nit: can we move this an individual function and add some comments for this if statement? It will be easier to read this if conditions.

Reply:
If you mean to only move the if condition in a function, I'm not sure if that would be much useful, as that is only used twice, at the places where we construct new UserNamespaceConfig. Can you please let me know if you meant to move anything else as well?
I'll definitely add comment to the condition explaining it in either case :+1:

Reply:
I mean `utils::rootless_required() && !utils::is_in_new_userns() && user_ns_config.is_none()` into a function. It will take a reader a good second to parse out all correct conditions to understand what this if statement does. I am fine with having some comments to explain.

Reply:
Hey, actually doing this helped me to test that function cleanly too! Thanks for the suggestion :+1: I have also added corresponding comment explaining the conditions.